tests/simplex/periodic_tet.cc

```diff
@@ -0,0 +1,300 @@
+                        << static_cast<unsigned int>(match.orientation)
+                        << " vs default "
+                        << static_cast<unsigned int>(default_orientation)
+                        << std::endl;
+                deallog << "Cells " << match.cell[0]->active_cell_index()
+                        << " and " << match.cell[1]->active_cell_index()
+                        << std::endl;
+                deallog << "Faces " << match.face_idx[0] << " and "
+                        << match.face_idx[1] << std::endl;
+                return;
+              }
+          }
+      }
+  }
+
+  {
+    // Check that we have gotten all the periodicity.
+    bool         mismatched_faces   = false;
+    unsigned int changing_direction = 0;
+    for (const auto &pair : pairs)
+      {
+        if (pair.size() != 8)
+          {
+            deallog << "FAIL! There should be 8 matches on each face."
+                    << std::endl;
+            return;
+          }
+        for (int face = 0; face < static_cast<int>(pair.size()); ++face)
+          {
+            Point<3> p1(
+              pair[face].cell[0]->face(pair[face].face_idx[0])->center());
+            Point<3> p2(
+              pair[face].cell[1]->face(pair[face].face_idx[1])->center());
+            std::array<const bool, 3> results = {
+              {std::abs(p1[0] - p2[0]) > 1e-10,
+               std::abs(p1[1] - p2[1]) > 1e-10,
+               std::abs(p1[2] - p2[2]) > 1e-10}};
+            bool fail = false;
+
+            for (unsigned int i = 0; i < 3; ++i)
+              {
+                if (i == changing_direction)
+                  {
+                    continue;
+                  }
+                else
+                  {
+                    fail = fail || results[i];
+                  }
+              }
+
+            if (fail)
+              {
+                deallog << "FAIL! Found non matching faces." << std::endl;
+                return;
+              }
+          }
+        ++changing_direction;
+      }
+  }
+
+  // write 2 outputs (total mesh and only surface mesh)
+  const auto grid_out = [](const auto &tria, const bool surface_mesh_only) {
+    GridOutFlags::Vtk flags;
+
+    if (surface_mesh_only)
+      {
+        flags.output_cells         = false;
+        flags.output_faces         = true;
+        flags.output_edges         = false;
+        flags.output_only_relevant = false;
+      }
+
+    GridOut grid_out;
+    grid_out.set_flags(flags);
+
+    grid_out.write_vtk(tria, deallog.get_file_stream());
+  };
+
+  grid_out(tria, false); // total mesh
+  grid_out(tria, true);  // only surface mesh
+
+  // If we got this far, we have a mesh which supports periodicity.
+  // Let's see if we can proscribe periodicity with actual DoFs now.
+  // Check for C0 tetrahedra up to degree 3. (Highest implemented at the time of
+  // writing.)
+  for (const unsigned int degree : {1, 2, 3})
+    {
+      DoFHandler     dof_handler(tria);
```

Comment:
```suggestion
      DoFHandler<3>     dof_handler(tria);
```
Our preferred style is that we don't use implicit templates unless they are required by specific generic programming techniques.